Python/pystate.c

```diff
@@ -293,8 +301,11 @@ _PyInterpreterState_DeleteExceptMain()
         if (interp->id_mutex != NULL) {
             PyThread_free_lock(interp->id_mutex);
         }
-        PyMem_RawFree(interp);
+        tmp_interp = interp;
```

Comment:
You can declare "PyInterpreterState *tmp_interp = NULL;" here. I suggest to rename it "prev_interp".